bundles/config/org.eclipse.smarthome.config.discovery.usbserial/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/internal/UsbSerialDiscoveryService.java

```diff
@@ -101,6 +109,7 @@ protected synchronized void unsetUsbSerialDiscovery(UsbSerialDiscovery usbSerial
         usbSerialDiscovery.stopBackgroundScanning();
         usbSerialDiscovery.unregisterDiscoveryListener(this);
         this.usbSerialDiscovery = null;
+        this.previouslyDiscovered.clear();
```

Comment:
Are you sure that you want to clear the Set if you remove **one** participant? I think you have to leave the Set untouched if you remove a participant and only remove items from it in `usbSerialDeviceRemoved`, right?

Reply:
@triller-telekom This method is called when  unsetting the `usbSerialDiscovery`, not when removing a `discoveryParticipant`.

Reply:
Github fooled me on this one... It folded away all lines between `protected void removeUsbSerialDiscoveryParticipant(UsbSerialDiscoveryParticipant participant) {` and `usbSerialDiscovery.stopBackgroundScanning();` so I did not see that your added code ended up in `unsetUsbSerialDiscovery`. To me it looked as if you added it to `removeUsbSerialDiscoveryParticipant`.

So all is good now :)